weather_sp/splitter_pipeline/file_splitters.py

```diff
@@ -147,37 +148,35 @@ def split_data(self) -> None:
                                   for var in dataset.data_vars])
             else:
                 iterlists.append([dataset])
-            for dim in ('time', 'level'):
-                if dim in self.output_info.split_dims():
-                    iterlists.append(dataset[dim])
+            filtered_split_dims = [
+                x for x in self.output_info.split_dims() if x not in ('variable', self._UNSUPPORTED_DIMENSIONS)]
+            for dim in filtered_split_dims:
+                iterlists.append(dataset[dim])
             combinations = itertools.product(*iterlists)
             for comb in combinations:
                 selected = comb[0]
                 for da in comb[1:]:
-                    if 'time' in da.coords:
-                        selected = selected.sel(time=da.time)
-                    if 'level' in da.coords:
-                        selected = selected.sel(level=da.level)
-                self._write_dataset(selected)
-            self.logger.info('split %s into %d files',
-                             self.input_path, len(list(combinations)))
+                    for dim in da.coords:
+                        selected = selected.sel({dim: getattr(da, dim)})
+                self._write_dataset(selected, filtered_split_dims)
+            self.logger.info('Finished splitting %s', self.input_path)
 
     @contextmanager
     def _open_dataset_locally(self) -> t.Iterator[xr.Dataset]:
         with self._copy_to_local_file() as local_file:
-            yield xr.open_dataset(local_file.name)
+            yield xr.open_dataset(local_file.name, engine='netcdf4')
 
-    def _write_dataset(self, dataset: xr.Dataset()) -> None:
-        with FileSystems().create(self._get_output_for_dataset(dataset)) as dest_file:
+    def _write_dataset(self, dataset: xr.Dataset(), split_dims: t.List[str]) -> None:
```

Comment:
Missed it last time: the type should be `xr.Dataset` I believe. Currently, the type is an instance of a dataset, not the type.

Reply:
oh wow! Nice catch! I had to read this three times to even notice what you meant!